src/openms/source/ANALYSIS/MAPMATCHING/TransformationModelLinear.cpp

```diff
@@ -44,6 +44,21 @@ namespace OpenMS
   {
     params_ = params;
     data_given_ = !data.empty();
+    // set x_weight
+    params_.setValue("x_weight",params.exists("x_weight") ? (std::string)params.getValue("x_weight") : "");
```

Comment:
Ok I think this part can be improved and might result in the failing tests.
You should check if x_weight exists and is not the empty string.
Only then store the meta values in param_
same for y_weight and the other models.
That way you are backwards compatible to the old test files.

Reply:
That makes complete sense.